CONTINUOUS_INTEGRATION.md

```diff
@@ -0,0 +1,194 @@
+ 4 -rwxrwxr-x  1 root root   538 Apr 20 22:52 host-build.sh
+ 4 -rwxrwxr-x  1 root root   556 Apr 20 22:52 node-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   343 Apr 20 22:52 node-tests.sh
+ 4 -rwxrwxr-x  1 root root   591 Apr 20 22:52 node_tflite-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   482 Apr 20 22:52 node_tflite-tests.sh
+ 4 -rwxrwxr-x  1 root root   614 Apr 20 22:52 package.sh
+ 4 -rwxrwxr-x  1 root root  2787 Apr 20 22:52 package-utils.sh
+ 4 -rwxrwxr-x  1 root root   521 Apr 20 22:52 python-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   274 Apr 20 22:52 python-tests.sh
+ 4 -rwxrwxr-x  1 root root   502 Apr 20 22:52 python_tflite-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   413 Apr 20 22:52 python_tflite-tests.sh
+ 4 -rwxrwxr-x  1 root root  2688 Apr 20 22:52 tf-build.sh
+ 4 -rwxrwxr-x  1 root root  2384 Apr 20 22:52 tf-package.sh
+ 4 -rwxrwxr-x  1 root root  3371 Apr 20 22:52 tf-setup.sh
+12 -rwxrwxr-x  1 root root  8375 Apr 20 22:52 tf-vars.sh
+```
+
+### Runners
+
+A _Runner_ is a server that listens for available _Jobs_ to run. The _Runner_ runs each _Job_ one at a time and then reports the results back to GitHub. Here, we assume you will use the built-in _Runner_ on GitHub to perform continuous integration.
+
+This example assumes you have cloned DeepSpeech to the location:
+
+`https://github.com/{USERNAME}/DeepSpeech/`
+
+1. Navigate to your GitHub repository's Settings -> Actions. This should be located at `https://github.com/{USERNAME}/DeepSpeech/settings/actions`. It's a good idea for security to change the default Action Permissions from "Allow all actions" to "Allow local actions only".
+
+2. Next, navigate to your GitHub repository's Actions. This should be located at `https://github.com/{USERNAME}/DeepSpeech/actions`. You should see three _Workflows_ already set up. GitHub recognises these from the files stored in the `.github/workflows` directory. Remember from the [Events](#Events) section earlier in this tutorial that the _Workflows_ are set up to trigger on `pull_request` and `push` events that occur on the `master` branch of the code.
+
+3. So, to test that they are working, we need to make a change to the `DeepSpeech` repository that triggers one of those events. Here, will create a `push` event because it's easier to demonstrate. Navigate to the repository's Code listing. This should be at `https://github.com/{USERNAME}/DeepSpeech`. Add a file called `test-workflow.txt` containing dummy text and commit the change. This should trigger a `push` event, and the _Workflows_ will activate. You can monitor the _Workflows_ executing from the repository's Actions page. This is at `https://github.com/{USERNAME}/DeepSpeech/actions`.
+
+It's good to be aware that the first time that your GitHub Actions run, the _Workflow_ will take around four hours to execute. This is due to the TensorFlow build cache. Subsequent runs will be much shorter; typically we have seen them take just over an hour to execute. 
```

Comment:
I think it's good but it might be an opportunity to elaborate a bit on the why:
 - tensorflow builds takes ~3h on the hardware available
 - tensorflow  itself does not change between each PR except when changing tensorflow :)
 - github cache is limietd to 5GB for all repo and the size of just tensorflow would consume all of it
 - that's why we build a cache manually using artifacts
 - that's why we have the `get_cache_key` as well as `check_artifact_exists` actions ;)

Reply:
@lissyx thank you, super helpful. How does the section I've added now read?